admin_panel/users/viewsets/auth.py

```diff
@@ -0,0 +1,55 @@
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from rest_framework.authtoken.views import ObtainAuthToken
+from django.contrib.auth.models import User
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+
+from admin_panel.users.serializers.auth import CustomAuthTokenSerializer
+
+
+class CustomAuthTokenView(ObtainAuthToken):
+    """
+    API endpoint for user authentication and token generation
+    """
+    serializer_class = CustomAuthTokenSerializer
+
+    @swagger_auto_schema(
+        tags=['Authentication'],
+        operation_summary="Obtain auth token",
+        operation_description="Authenticate user and get token along with user details",
+        request_body=CustomAuthTokenSerializer,
+        responses={
+            200: openapi.Response(
+                description="Successful authentication",
+                schema=openapi.Schema(
+                    type=openapi.TYPE_OBJECT,
+                    properties={
+                        'token': openapi.Schema(type=openapi.TYPE_STRING),
+                        'user_id': openapi.Schema(type=openapi.TYPE_INTEGER),
+                        'organization': openapi.Schema(type=openapi.TYPE_INTEGER, nullable=True),
+                        'email': openapi.Schema(type=openapi.TYPE_STRING),
+                    }
+                )
+            ),
+            400: "Invalid credentials"
+        }
+    )
+    def post(self, request, *args, **kwargs):
+        serializer = CustomAuthTokenSerializer(
+            data=request.data,
+            context={'request': request}
+        )
+        serializer.is_valid(raise_exception=True)
+        email = serializer.validated_data['email']
+        user = User.objects.filter(email=email).first()
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding user existence validation</b></div>
<div id="fix">

Consider adding validation to check if user exists before accessing user properties. The current code might raise errors if no user is found with the given email. A similar issue was also found in admin_panel/viewsets/user/auth.py (line 18-19).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -44,6 +44,9 @@
         email = serializer.validated_data['email']
         user = User.objects.filter(email=email).first()
 +       if not user:
 +           return Response({'error': 'User not found'}, status=400)
 +
         token, created = Token.objects.get_or_create(user=user)
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/isshiki-examevokee/ExamEvoke/pull/2#issuecomment-2496536252>#ae3661</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged